en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-300-to-310.md

```diff
@@ -0,0 +1,713 @@
+        2. Go to the `<API-M_ANALYTICS_HOME>/bin` directory. Run the command given below, to generate the converted file in the `<API-M_ANALYTICS_HOME>/lib` directory.
+
+        ```
+        ./jartobundle.sh <PATH_TO_NON-OSGi_JAR> ../lib
+        ```
+
+5.  Start the Worker and Dashboard profiles as below by navigating to `<API-M_ANALYTICS_3.1.0_HOME>/bin` location.
+    
+    ```tab="Worker"
+    sh worker.sh
+    ```
+
+    ```tab="Dashboard"
+    sh dashboard.sh
+    ```
+
+!!! note
+    If you have developed any custom dashboards in API-M 3.0.0 Analytics using Stream Processor, you will be able to use the same in API-M Anaytics 3.1.0 as well. If you require any guidance regarding this, you can contact [WSO2 Support](https://support.wso2.com/jira/secure/Dashboard.jspa).
+
+#### Step 3.3 - Configure WSO2 API-M 3.1.0 for Analytics
+
+Follow the instructions below to configure WSO2 API Manager for the WSO2 API-M Analytics migration in order to migrate the statistics related data.
+
+1.  Configure the following datasources in the `<API-M_3.1.0_HOME>/repository/conf/deployment.toml` file.
+
+    The following is an example of how the configurations should be defined when using MySQL.
+
+    This datasource points to the **previous API-M version's WSO2AM_DB datasource.**
+
+    ``` java
+    [database.apim_db]
+    type = "mysql"
+    url = "jdbc:mysql://localhost:3306/am_db"
+    username = "username"
+    password = "password"
+    ```
+
+2.  Enable analytics in WSO2 API-M by setting the following configuration to true in the `<API-M_3.1.0_HOME>/repository/conf/deployment.toml` file.
+
+    ``` java
+    [apim.analytics]
+    enable = true
+    ```
+
+### Step 4 - Restart the WSO2 API-M 3.1.0 server
+
+1.  Restart the WSO2 API-M server.
+
+    ```tab="Linux / Mac OS"
+    sh wso2server.sh
+    ```
+
+    ```tab="Windows"
+    wso2server.bat
+    ```
+
+    !!! note "If you have enabled Analytics"
+        After starting the WSO2 API-M server and the WSO2 API-M Analytics 3.1.0 server from worker and dashboard profiles, the dashboards can be accessed via `https://<dashboard-server-host-name>:9643/analytics-dashboard` link.
+
+        !!! warning
+            Make sure you have started the API-M server node before accessing the Dashboard profile as the authentication happens via the API-M's authentication admin service.
+
+This concludes the upgrade process.
+
+!!! tip
+    The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
+
+!!! note
+    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 3.1.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
```

Comment:
Do we need this info? AFAIK, JWT authentication will work for the APIs migrated from 3.0.0.

Reply:
Yes this is correct. It can be useful when they have created OAUTH2 application in 3.0.0. But if they have created an OAUTH2 application in 3.0.0 they would want to continue using the APIs with OAUTH2 itself. Hence, resolved with 02d0d1b869f0f90576477aead17e8dfa5c919384